applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```
@@ -304,11 +304,18 @@ def get_linear_solver(self):
         return self._linear_solver
 
     def get_builder_and_solver(self):
+        if (self.settings["multi_point_constraints_used"].GetBool() is False and
```

I think the best would be if we don't need the parameter `"multi_point_constraints_used"`. But, if we do need it, it should be consistent with the use of multipoint constraints. Leaving it false in the ProjectParameters.json but converting it to true in the python is inconsistent and confusing in my opinion. I would throw an error instead.

---

I agree with this, how do you suggest to handle the trilinos version?
This one cannot handle constraints yet

Issue an Info if constraints are present?

---

Probably throw an error.

---

+1 for automatically detecting if constraints are there instead of having a parameters passed